common/variant/src/main/java/org/apache/spark/types/variant/VariantBuilder.java

```diff
@@ -223,7 +223,7 @@ public void appendFloat(float f) {
 
   public void appendBinary(byte[] binary) {
     checkCapacity(1 + U32_SIZE + binary.length);
-    writeBuffer[writePos++] = primitiveHeader(LONG_STR);
+    writeBuffer[writePos++] = primitiveHeader(BINARY);
```

Comment:
I think this change is already in master?

Reply:
Yes, I'll try to resolve it. I think I've made a bit of a mess with the master branch in my fork.

Reply:
Should be resolved now!